### PR TITLE
BTRX-289 Missing information in downloadable Full DAR Application PDF and BTRX-409 DUOS: DAR Full Application PDF - Research Use Statement cuts off at page bottom

### PR DESCRIPTION
**Pull request by nlopezzalba**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: consent-ws/src/main/java/org/broadinstitute/consent/http/resources/DataRequestReportsResource.java [hunk 1]
```diff
@@ -79,4 +89,34 @@ public Response downloadReviewedDARs() {
             return createExceptionResponse(e);
         }
     }
+
+    private Boolean requiresManualReview(Document dar) throws IOException {
+        Map<String, Object> form = parseAsMap(dar.toJson());
```
Comment (rushtong) [inline consent-ws/src/main/java/org/broadinstitute/consent/http/resources/DataRequestReportsResource.java]: Not for this PR, but we should keep this in mind for future work. We parse this as a map in lots of places, looking for particular values. I think this kind of code should be in a utility class specifically designed around parsing out `dar` documents.

File: consent-ws/src/main/java/org/broadinstitute/consent/http/resources/DataRequestReportsResource.java [hunk 2]
```diff
@@ -79,4 +89,34 @@ public Response downloadReviewedDARs() {
...
+        }
+        return false;
+    }
+
+    private Map<String, Object> parseAsMap(String str) throws IOException {
+        ObjectReader reader = mapper.readerFor(Map.class);
+        return reader.readValue(str);
+    }
+
+    private String[] fieldsForManualReview = {
```
Comment (rushtong) [inline consent-ws/src/main/java/org/broadinstitute/consent/http/resources/DataRequestReportsResource.java]: This is the same in `DataAccessRequestResource`. We should refactor both into a single array

File: consent-ws/src/main/java/org/broadinstitute/consent/http/service/DataAccessParser.java [hunk 3]
```diff
@@ -194,6 +195,43 @@ public PDAcroForm fillDARForm(Document dar, Map<String, String> researcherProper
                     field.setValue(getYesOrNoValue(dar.getBoolean(DarConstants.POA)));
                     break;
                 }
+                case DarConstants.CHECK_COLLABORATOR: {
+                    field.setValue(getYesOrNoValue(dar.getBoolean(DarConstants.CHECK_COLLABORATOR)));
+                    break;
+                }
+                case "userStatus": {
```
Comment (rushtong) [inline consent-ws/src/main/java/org/broadinstitute/consent/http/service/DataAccessParser.java]: Any reason these case strings (here and below) are not a constant like we have for other cases?

File: consent-ws/src/main/java/org/broadinstitute/consent/http/service/DataAccessParser.java [hunk 4]
```diff
@@ -194,6 +195,43 @@ public PDAcroForm fillDARForm(Document dar, Map<String, String> researcherProper
...
+                    field.setValue(getDefaultValue(StringUtils.capitalize(role.getStatus())));
+                    break;
+                }
+                case "adminComment": {
+                    field.setValue(getDefaultValue(StringUtils.capitalize(role.getRationale())));
```
Comment (rushtong) [inline consent-ws/src/main/java/org/broadinstitute/consent/http/service/DataAccessParser.java:207]: Just double checking that `role.getRationale()` is really the `adminComment` case.

**Review by rushtong — CHANGES_REQUESTED** [review]
See comments inline - the one thing I'd like to see is pulling out the required field names into a common object. Other than that, just some questions.

File: consent-ws/src/main/java/org/broadinstitute/consent/http/service/DataAccessParser.java (hunk 4 above)
Comment (vvicario) [reply consent-ws/src/main/java/org/broadinstitute/consent/http/service/DataAccessParser.java:207]: yes, rational and comments are saved in the same place

**Comment by vvicario** [comment]
@rushtong I worked on your feedback. Please let me know if you have any other comment

**Review by rushtong — APPROVED** [review]
👍

**Pull request merged**